src/runtime/runtime.ts

```diff
@@ -221,12 +253,81 @@ export class Runtime {
     // we could eliminate it if the Manifest object takes care of this.
     const id = `in-memory-${Math.floor((Math.random()+1)*1e6)}.manifest`;
     // TODO(sjmiles): this is a virtual manifest, the fileName is invented
-    const localOptions = {id, fileName: `./${id}`, loader};
-    return Manifest.parse(content, {...localOptions, ...options});
+    const opts = {id, fileName: `./${id}`, loader, ...options};
+    return Manifest.parse(content, opts);
+  }
+
+  async parseFile(path: string, options?): Promise<Manifest> {
```

Comment:
Can you add a type for the `options` param?

Reply:
I remember deferring this work too, but I can't remember the reason, so I'll go back and check.

Reply:
Tracked in #3997